File: tmdb-core/src/main/java/dev/reuss/tmdb/value/id/MovieId.java
package dev.reuss.tmdb.value.id;

/**
 * TMDB movie id.
 *
 * @param value TMDB movie id
 */
public record MovieId(int value) implements NumericTmdbResourceId {

    public MovieId {
        if (value <= 0) {
            throw new IllegalArgumentException("Movie id must be greater than 0");
        }
    }

    public static MovieId of(int value) {
        return new MovieId(value);
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/id/MovieId.kt
package dev.reuss.tmdb.value.id

/**
 * TMDB movie id.
 *
 * @property value TMDB movie id
 */
class MovieId private constructor(
    value: Int,
) : NumericTmdbResourceId(value, "Movie id") {
    companion object {
        /**
         * Creates an identifier from a positive TMDB resource ID.
         *
         * @throws IllegalArgumentException if [value] is not greater than `0`
         */
        @JvmStatic
        fun of(value: Int): MovieId = MovieId(value)
    }
}
